### PR TITLE
Headless/tahsin/compare/respone update

### PR DESCRIPTION
**Pull request by ghost**
# Description

(Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.)

## Type of change

(Please delete options that are not relevant.)

- [x] Link to a **GH Issue** or **Jira** task here for reference.
- [ ] Bug fix
- [x] New feature

# How Has This Been Tested?

(Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration)

- [x] Tested by myself
- [ ] Unit test

# Developer Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Review check list:

- [x] Separation of Concerns followed (Reviewer name/NaN)
- [x] Code is in sync with existing code patterns/technologies. (Reviewer name/NaN)
- [ ] Are there any obvious optimizations that will improve performance? (Reviewer name/NaN)
- [ ] Can any logging or debugging code be removed? (Reviewer name/NaN)
- [ ] Are all data inputs checked (for the correct type, length, format, and range) and encoded? (Reviewer name/NaN)
- [ ] Are invalid parameter values handled? (Reviewer name/NaN)
- [ ] Is error handling done the correct way? (Reviewer name/NaN)
- [ ] Are all the edge cases (null, 0, negative) handled properly? (Reviewer name/NaN)

### CODE REVIEW

**Comment by ghost** [comment]
[https://pm23.atlassian.net/browse/BC-579](url)

File: packages/headless/src/entity/compare.ts [hunk 1]
```diff
@@ -1,11 +1,19 @@
 export class IProductInfo {
   name: string;
   price: number;
+  oldPrice: number;
   shortDescription: string;
   fullDescription: string;
 }
+export class IProductMeta{
```
Comment (ismail-bs) [inline packages/headless/src/entity/compare.ts:8]: We need just one property(friendlyPageName).

File: packages/headless/src/modules/compare/dto/compare.dto.ts [hunk 2]
```diff
@@ -18,20 +21,42 @@ export class AddToCompareDto implements AddCompareItem {
...
+  @ApiProperty({ required: false })
+  @IsOptional()
+  @IsString()
+  title?: string;
+
+  @ApiProperty({ required: false })
+  @IsOptional()
+  @IsString()
+  description?: string;
+}
```
Comment (ismail-bs) [inline packages/headless/src/modules/compare/dto/compare.dto.ts:43]: friendlyPageName is missing.

File: packages/headless/src/modules/compare/dto/compare.dto.ts [hunk 3]
```diff
@@ -18,20 +21,42 @@ export class AddToCompareDto implements AddCompareItem {
...
+export class ProductDetails implements IProductDetails{
   @ApiProperty({ type: () => [ProductInfo] })
   info: ProductInfo;
 
+  @ApiProperty({ type: () => [IProductMetaDto] })
```
Comment (ismail-bs) [inline packages/headless/src/modules/compare/dto/compare.dto.ts]: why this is an array?

File: packages/headless/src/entity/compare.ts (hunk 1 above)
Comment (ghost) [reply packages/headless/src/entity/compare.ts:8]: please elaborate

**Pull request merged**